src/types/PlaidTypes.res

```diff
@@ -0,0 +1,257 @@
+  token: string,
+  noLoadingState?: bool,
+  ...commonPlaidLinkOptions,
+}
+
+type linkErrorCode =
+  // ITEM_ERROR
+  | INVALID_CREDENTIALS
+  | INVALID_MFA
+  | ITEM_LOGIN_REQUIRED
+  | INSUFFICIENT_CREDENTIALS
+  | ITEM_LOCKED
+  | USER_SETUP_REQUIRED
+  | MFA_NOT_SUPPORTED
+  | INVALID_SEND_METHOD
+  | NO_ACCOUNTS
+  | ITEM_NOT_SUPPORTED
+  | TOO_MANY_VERIFICATION_ATTEMPTS
+  | INVALID_UPDATED_USERNAME
+  | INVALD_UPDATED_USERNAME
+  | ITEM_NO_ERROR
+  | ITEM_NO_ERROR_LOWERCASE
+  | NO_AUTH_ACCOUNTS
+  | NO_INVESTMENT_ACCOUNTS
+  | NO_INVESTMENT_AUTH_ACCOUNTS
+  | NO_LIABILITY_ACCOUNTS
+  | PRODUCTS_NOT_SUPPORTED
+  | ITEM_NOT_FOUND
+  | ITEM_PRODUCT_NOT_READY
+  // INSTITUTION_ERROR
+  | INSTITUTION_DOWN
+  | INSTITUTION_NOT_RESPONDING
+  | INSTITUTION_NOT_AVAILABLE
+  | INSTITUTION_NO_LONGER_SUPPORTED
+  // API_ERROR
+  | INTERNAL_SERVER_ERROR
+  | PLANNED_MAINTENANCE
+  // ASSET_REPORT_ERROR
+  | PRODUCT_NOT_ENABLED
+  | DATA_UNAVAILABLE
+  | ASSET_PRODUCT_NOT_READY
+  | ASSET_REPORT_GENERATION_FAILED
+  | INVALID_PARENT
+  | INSIGHTS_NOT_ENABLED
+  | INSIGHTS_PREVIOUSLY_NOT_ENABLED
+  // BANK_TRANSFER_ERROR
+  | BANK_TRANSFER_LIMIT_EXCEEDED
+  | BANK_TRANSFER_MISSING_ORIGINATION_ACCOUNT
+  | BANK_TRANSFER_INVALID_ORIGINATION_ACCOUNT
+  | BANK_TRANSFER_ACCOUNT_BLOCKED
+  | BANK_TRANSFER_INSUFFICIENT_FUNDS
+  | BANK_TRANSFER_NOT_CANCELLABLE
+  | BANK_TRANSFER_UNSUPPORTED_ACCOUNT_TYPE
+  | BANK_TRANSFER_UNSUPPORTED_ENVIRONMENT
+  // SANDBOX_ERROR
+  | SANDBOX_PRODUCT_NOT_ENABLED
+  | SANDBOX_WEBHOOK_INVALID
+  | SANDBOX_BANK_TRANSFER_EVENT_TRANSITION_INVALID
+  // INVALID_REQUEST
+  | MISSING_FIELDS
+  | UNKNOWN_FIELDS
+  | INVALID_FIELD
+  | INCOMPATIBLE_API_VERSION
+  | INVALID_BODY
+  | INVALID_HEADERS
+  | NOT_FOUND
+  | NO_LONGER_AVAILABLE
+  | SANDBOX_ONLY
+  | INVALID_ACCOUNT_NUMBER
+  // INVALID_INPUT
+  // From above ITEM_LOGIN_REQUIRED = "INVALID_CREDENTIALS",
+  | INCORRECT_DEPOSIT_AMOUNTS
+  | UNAUTHORIZED_ENVIRONMENT
+  | INVALID_PRODUCT
+  | UNAUTHORIZED_ROUTE_ACCESS
+  | DIRECT_INTEGRATION_NOT_ENABLED
+  | INVALID_API_KEYS
+  | INVALID_ACCESS_TOKEN
+  | INVALID_PUBLIC_TOKEN
+  | INVALID_LINK_TOKEN
+  | INVALID_PROCESSOR_TOKEN
+  | INVALID_AUDIT_COPY_TOKEN
+  | INVALID_ACCOUNT_ID
+  | MICRODEPOSITS_ALREADY_VERIFIED
+  // INVALID_RESULT
+  | PLAID_DIRECT_ITEM_IMPORT_RETURNED_INVALID_MFA
+  // RATE_LIMIT_EXCEEDED
+  | ACCOUNTS_LIMIT
+  | ADDITION_LIMIT
+  | AUTH_LIMIT
+  | BALANCE_LIMIT
+  | IDENTITY_LIMIT
+  | ITEM_GET_LIMIT
+  | RATE_LIMIT
+  | TRANSACTIONS_LIMIT
+  // RECAPTCHA_ERROR
+  | RECAPTCHA_REQUIRED
+  | RECAPTCHA_BAD
+  // OAUTH_ERROR
+  | INCORRECT_OAUTH_NONCE
+  | OAUTH_STATE_ID_ALREADY_PROCESSED
+
+type linkErrorType =
+  | BANK_TRANSFER_ERROR
+  | INVALID_REQUEST
+  | INVALID_RESULT
+  | INVALID_INPUT
+  | INSTITUTION_ERROR
+  | RATE_LIMIT_EXCEEDED
+  | API_ERROR
+  | ITEM_ERROR
+  | AUTH_ERROR
+  | ASSET_REPORT_ERROR
+  | SANDBOX_ERROR
+  | RECAPTCHA_ERROR
+  | OAUTH_ERROR
+
+type linkExitMetadataStatus =
```

Comment:
Where it is being use? Can we remove it?

Reply:
removed these error types 👍